lib/core/base/cache.js

```diff
@@ -7,16 +7,29 @@ const cache = {
    * @param {*} value - Value to store.
    */
   set(key, value) {
+    validateKey(key);
+
     _cache[key] = value;
   },
 
   /**
    * Retrieve an item from the cache.
    * @param {String} key - Name of the key the value was stored as.
+   * @param {Function} [creator] - Default value to set if there is a cache miss. Functions are evaluated before caching. To override a value already saved, use `set()`.
    * @returns {*} The item stored
    */
-  get(key) {
-    return _cache[key];
+  get(key, creator) {
+    validateCreator(creator);
+
+    if (key in _cache) {
+      return _cache[key];
+    }
+
+    if (typeof creator === 'function') {
+      const value = creator();
```

Comment:
Did we not say we were going to do this?

```suggestion
      const value = creator();
      assert(value !== 'undefined', 'Cache creator function should not return undefined');
```

Reply:
Without the extra quotes:

```suggestion
      const value = creator();
      assert(value !== undefined, 'Cache creator function should not return undefined')
```

Reply:
Asked on Slack (but didn't get a response), so I'll do it again here...

If we are not allowing:

```js
cache.get('foo', function(){ return undefined })
```

Then why do we allow the below?

```js
cache.set('foo', undefined)
```

Reply:
The assert helper has been integrated. Stephen, I think Steven got to chat with you on Slack last week. Thanks for the feedback, but it looks like we're going anther direction on caching undefined so I'm going to resolve this conversation to get the PR ready for re-review with current scope.

Reply:
The question above was never answered (neither on Slack nor here).